generator/transit_generator.cpp

```diff
@@ -1,16 +1,146 @@
+/// For example,
+/// GetFileName("Russia_Nizhny Novgorod Oblast.transit.json") returns "Russia_Nizhny Novgorod Oblast"
+/// GetFileName("Russia_Nizhny Novgorod Oblast.mwm") returns "Russia_Nizhny Novgorod Oblast"
+string GetFileName(string const & filePath)
+{
+  string name = filePath;
+  my::GetNameFromFullPath(name);
+
+  string nameWithExt;
+  do
+  {
+    nameWithExt = name;
+    my::GetNameWithoutExt(name);
+  }
+  while (nameWithExt != name);
+
+  return name;
+}
+
+/// \brief Reads from |root| (json) and serialize an array to |serializer|.
+template <class Item>
+void SerializeObject(my::Json const & root, string const & key, Serializer<FileWriter> & serializer)
+{
+  vector<Item> items;
+
+  DeserializerFromJson deserializer(root.get());
+  deserializer(items, key.c_str());
+
+  for (auto const & s : items)
+    s.Visit(serializer);
+}
+}  // namespace
+
 namespace routing
 {
 namespace transit
 {
-void BuildTransit(std::string const & mwmPath, std::string const & transitDir)
+// DeserializerFromJson ---------------------------------------------------------------------------
+void DeserializerFromJson::operator()(m2::PointD & p, char const * name)
 {
-  LOG(LINFO, ("mwm path:", mwmPath, ", directory with transit:", transitDir));
+  json_t * pointItem = my::GetJSONOptionalField(m_node, name);
+  if (pointItem == nullptr)
+    return;
+  CHECK(json_is_object(pointItem), ());
+  FromJSONObject(pointItem, "x", p.x);
+  FromJSONObject(pointItem, "y", p.y);
+}
+
+void BuildTransit(string const & mwmPath, string const & transitDir)
+{
+  LOG(LERROR, ("This method is under construction and should not be used for building production mwm "
+      "sections."));
   NOTIMPLEMENTED();
+
+  string const countryId = GetFileName(mwmPath);
+  LOG(LINFO, ("countryId:", countryId));
+
+  string const graphFullPath = my::JoinFoldersToPath(transitDir, countryId + ".transit.json");
+
+  Platform::EFileType fileType;
+  Platform::EError const errCode = Platform::GetFileType(graphFullPath, fileType);
+  if (errCode != Platform::EError::ERR_OK || fileType != Platform::EFileType::FILE_TYPE_REGULAR)
+  {
+    LOG(LINFO, ("For mwm:", mwmPath, ".transit.json file not found"));
+    return;
+  }
+
+  // @TODO(bykoianko) In the future transit edges which cross mwm border will be split in the generator. Then
+  // routing will support cross mwm transit routing. In current version every json with transit graph
+  // should have a special name: <country id>.transit.json.
+
+  LOG(LINFO, ("Creating", TRANSIT_FILE_TAG, "section for:", countryId, "based on", graphFullPath));
+
+  string jsonBuffer;
+  try
+  {
+    GetPlatform().GetReader(graphFullPath)->ReadAsString(jsonBuffer);
+  }
+  catch (RootException const & ex)
+  {
+    LOG(LCRITICAL, ("Can't open", graphFullPath, ex.what()));
+  }
+
+  // @TODO(bykoianko) If it's necessary to parse an integer jansson parser keeps it to time long long value.
```

Comment:
Sorry, I didn't get the first sentence.

Reply:
It contained mistakes. It should be: `  // @TODO(bykoianko) If it's necessary to parse an integer jansson parser keeps it in long long value.`